util/cmake/SeqAnBuildSystem.cmake

```diff
@@ -247,14 +254,25 @@ macro (seqan_build_system_init)
         if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
             set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -ipo -no-prec-div -fp-model fast=2 -xHOST")
         endif ()
-    elseif (SEQAN_ARCH_SSE4)
-        message (STATUS "Building optimized binaries up to SSE4 and POPCNT.")
-        set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -mmmx -msse -msse2 -msse3 -mssse3 -msse4")
-        if (NOT ${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
-            set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -mpopcnt")
-        endif ()
-        if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
-            set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -ipo -no-prec-div -fp-model fast=2")
+    else ()
+        if (SEQAN_ARCH_SSE4)
+            if (SEQAN_ARCH_AVX2)
+                message (STATUS "Building optimized binaries up to AVX2 and POPCNT.")
+                set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -mavx -march=core-avx2")
```

Comment:
`-march=` are **architecture** flags, they imply a **set** of flags for a **specific CPU (generation)**
In this case `-march=core-avx2` is the equivalent of `-march=haswell` which specifically target intel haswell (or newer) CPUS, activating the following features:
> MOVBE, MMX, SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2, POPCNT, AVX, AVX2, AES, PCLMUL, FSGSBASE, RDRND, FMA, BMI, BMI2 and F16C instruction set support.

which could as well be activated manually.... but which we don't want to do, because it will break on newer AMDs (e.g. newest bulldozer which support AVX2, but don't support RDRND).

→ please always use explicit `-m*` flags and not `-march=*` because the latter bring a lot of other stuff which we don't know and which will probably not work on other platforms.